.github/workflows/build-and-deploy.yaml

```diff
@@ -0,0 +1,117 @@
+name: build and deploy
+
+on:
+  push:
+    branches:
+      - '**'
```

Comment:
So actually the reason we don't get an infinite loop is because the `deploy` job has a condition that it'll only run if the changed services matrix isn't empty. When "deploy" it, the CI pushes changes only to the `/manifests` directory and isn't updating any services, so on that push the service matrix will be an empty list and the deployment won't run.

A bit confusing, but I'll make sure that gets documented